src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -294,37 +291,62 @@ private static bool IsValueTypeErrorOrBlank<T>(FormulaValue val)
             return new InMemoryTableValue(irContext, pairs.Select(pair => pair.Key));
         }
 
-        private static async IAsyncEnumerable<DValue<RecordValue>> LazyFilterAsync(
+        private static async Task<DValue<RecordValue>> LazyFilterRowAsync(
+           EvalVisitor runner,
+           SymbolContext context,
+           DValue<RecordValue> row,
+           LambdaFormulaValue filter)
+        {
+            if (row.IsValue)
+            {
+                var childContext = context.WithScopeValues(row.Value);
+
+                // Filter evals to a boolean 
+                var result = await filter.EvalAsync(runner, childContext);
+                var include = false;
+                if (result is BooleanValue booleanValue)
+                {
+                    include = booleanValue.Value;
+                }
+                else if (result is ErrorValue errorValue)
+                {
+                    return DValue<RecordValue>.Of(errorValue);
+                }
+
+                if (include)
+                {
+                    return row;
+                }
+            }
+
+            return null;
+        }
+
+        private static async Task<DValue<RecordValue>[]> LazyFilterAsync(
```

Comment:
>Task<DValue<RecordValue>[]> [](http://example.com/codeflow?start=29&length=27)

Instead of returning a (task of) array, can we return a task of an enumerable? This way something like `LookUp([0, 10], Async(Value) >= 0)` would complete (today I'm guessing it would hang)